roles/ceph-config/tasks/main.yml

```diff
@@ -107,22 +69,23 @@
             mode=0644
   register: result_ceph_conf
 
-- name: set ceph restart flag for ceph conf change
-  copy: dest=/etc/ansible/facts.d/restart_ceph.fact content='true'
-  when: result_ceph_conf.changed
+# set restart flag for ceph in case we fail before restarting ceph
+# we dont' restart ceph for new osd node, because that will block scale-out
+- block:
+  - name: set ceph restart flag
+    copy: dest=/etc/ansible/facts.d/restart_ceph.fact content='true'
 
-- name: re-read facts after adding custom fact
-  setup: filter=ansible_local
-  when: result_ceph_conf.changed
+  - name: re-read facts after adding custom fact
+    setup: filter=ansible_local
+  when:
+    - "'ceph_monitors' in group_vars or 'ceph_osds' in group_vars" 
+    - stat_ceph_conf.stat.exists
+    - result_ceph_conf.changed
 
 - include: logging.yml
-  tags:
-    - logrotate
-    - logging
-  when: logging.enabled|default('True')|bool
+  tags: ['logrotate', 'logging']
+  when: logging.enabled|default(true)
 
 - include: serverspec.yml
```

Comment:
is ceph_spec getting installed on controller/computes (ceph client systems?) if so we will need to adjust the serverspecs as lot of logs files it's is looking for will not be present on compute/controller